packages/react/src/components/LinkWithIcon/LinkWithIcon.js

```diff
@@ -17,33 +23,120 @@ const { prefix } = settings;
 /**
  * LinkWithIcon component.
  */
-const LinkWithIcon = ({ children, href, ...props }) => {
+const LinkWithIcon = ({
+  children,
+  href,
+  copy,
+  type,
+  direction,
+  visited,
+  ...props
+}) => {
+  const RenderIcon = _renderLinkIcon(type);
+
+  console.log('visited state', visited);
   return (
     <div
-      className={`${prefix}--link-with-icon__container`}
+      className={cx(`${prefix}--link-with-icon__container`, {
+        [`${prefix}--link-with-icon__container--disabled`]: props.disabled,
+      })}
       data-autoid={`${stablePrefix}--link-with-icon`}>
-      <Link href={href} className={`${prefix}--link-with-icon`} {...props}>
+      <Link
```

Comment:
This doesn't look right. `LinkWithIcon` is meant to allow any kind of icons here, and not restricting it to the 4 you have imported. This is a generic component.